tests/test_numeric/test_eulerian_grid_ops/test_poisson_solver_3d/test_unbounded_poisson_solver_mpi_3d.py

```diff
@@ -0,0 +1,213 @@
+    ghost_size, precision, rank_distribution, aspect_ratio
+):
+    n_values = 8
+    grid_size_z, grid_size_y, grid_size_x = (n_values * np.array(aspect_ratio)).astype(
+        int
+    )
+    real_t = get_real_t(precision)
+    # Generate the MPI topology minimal object
+    mpi_construct = MPIConstruct3D(
+        grid_size_z=grid_size_z,
+        grid_size_y=grid_size_y,
+        grid_size_x=grid_size_x,
+        real_t=real_t,
+        rank_distribution=rank_distribution,
+    )
+
+    # Create unbounded poisson solver
+    unbounded_poisson_solver = UnboundedPoissonSolverMPI3D(
+        grid_size_z=grid_size_z,
+        grid_size_y=grid_size_y,
+        grid_size_x=grid_size_x,
+        real_t=real_t,
+        mpi_construct=mpi_construct,
+        ghost_size=ghost_size,
+    )
+
+    # Initialize communicator for scatter and gather
+    mpi_field_comm = MPIFieldCommunicator3D(
+        ghost_size=ghost_size, mpi_construct=mpi_construct
+    )
+    gather_local_field = mpi_field_comm.gather_local_field
+    scatter_global_field = mpi_field_comm.scatter_global_field
+
+    # Allocate local field
+    local_rhs_vector_field = np.zeros(
+        (
+            mpi_construct.grid_dim,
+            mpi_construct.local_grid_size[0] + 2 * ghost_size,
+            mpi_construct.local_grid_size[1] + 2 * ghost_size,
+            mpi_construct.local_grid_size[2] + 2 * ghost_size,
+        )
+    ).astype(real_t)
+    local_solution_vector_field = np.zeros_like(local_rhs_vector_field)
+
+    # Initialize and broadcast solution for comparison later
+    if mpi_construct.rank == 0:
+        ref_rhs_vector_field = np.random.rand(
+            mpi_construct.grid_dim, grid_size_z, grid_size_y, grid_size_x
+        ).astype(real_t)
+    else:
+        ref_rhs_vector_field = (None, None, None)
+
+    # scatter global field
+    # TODO: replace with vector scatter when available
+    scatter_global_field(
+        local_rhs_vector_field[0], ref_rhs_vector_field[0], mpi_construct
+    )
+    scatter_global_field(
+        local_rhs_vector_field[1], ref_rhs_vector_field[1], mpi_construct
+    )
+    scatter_global_field(
+        local_rhs_vector_field[2], ref_rhs_vector_field[2], mpi_construct
+    )
+
+    # compute the unbounded poisson solve
+    unbounded_poisson_solver.vector_field_solve(
+        solution_vector_field=local_solution_vector_field,
+        rhs_vector_field=local_rhs_vector_field,
+    )
+
+    # gather back the solution field globally
+    global_solution_vector_field = np.zeros_like(ref_rhs_vector_field)
+    # TODO: replace with vector gather when available
+    gather_local_field(
+        global_solution_vector_field[0], local_solution_vector_field[0], mpi_construct
+    )
+    gather_local_field(
+        global_solution_vector_field[1], local_solution_vector_field[1], mpi_construct
+    )
+    gather_local_field(
+        global_solution_vector_field[2], local_solution_vector_field[2], mpi_construct
+    )
+
```

Comment:
same.

Reply:
Similarly fixed in [c5a5c41](https://github.com/fankiat/sopht-mpi/pull/135/commits/c5a5c4170b4659e61254f22d4080f5b62c41172a)